### PR TITLE
🔥 Do not format the messages file for `gen-l10n`

### LINKED ISSUE #119594: [gen-l10n] Should not use `dart format` for the untranslated message file
...because it cannot be formatted by `dart format`, which will cause the generation failed.

This also blocks users that used to have untranslated fields after they upgraded to Flutter 3.7.

image

```
[!] Flutter (Channel stable, 3.7.0, on Microsoft Windows [版本 10.0.22000.1455], locale zh-CN)
    • Flutter version 3.7.0 on channel stable at X:\SDK\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision b06b8b2710 (7 days ago), 2023-01-23 16:55:55 -0800
    • Engine revision b24591ed32
    • Dart version 2.19.0
    • DevTools version 2.20.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn
```

### PR DESCRIPTION
**Pull request by AlexV525**
Resolves #119594.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart [hunk 1]
```diff
@@ -265,4 +265,56 @@ format: true
...
+      artifacts: artifacts,
+      processManager: processManager,
+    );
+    await createTestCommandRunner(command).run(<String>['gen-l10n']);
+
+    final Directory outputDirectory = fileSystem.directory(fileSystem.path.join('.dart_tool', 'flutter_gen', 'gen_l10n'));
+    expect(outputDirectory.existsSync(), true);
+    expect(outputDirectory.childFile('app_localizations_en.dart').existsSync(), true);
+    expect(outputDirectory.childFile('app_localizations_es.dart').existsSync(), true);
+    expect(outputDirectory.childFile('app_localizations.dart').existsSync(), true);
```
[the remark this answers is not in the document]
Comment (AlexV525) [reply packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart:314]: It actually does exist, but in the `MemoryFileSystem`, the parsed file URI is incorrect which will cause the generated file does not locate under the correct directory, but instead as a file that contains path separators:
image

So the expectation is always false on Windows in this case.

The file URI was produced by the below line:
https://github.com/flutter/flutter/blob/6f9a896d7928970338ad0655feee09feff6d27c0/packages/flutter_tools/lib/src/localizations/gen_l10n.dart#L66

@jonahwilliams might know something about the path issue across host platforms.

[the remark this answers is not in the document]
Comment (AlexV525) [reply packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart:314]: I've added a replacement during parsing the file from the path by replacing all `\` to the separator from the file system. Let's see if it works on all platforms.

[the remark this answers is not in the document]
Comment (christopherfujino) [reply packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart:314]: When constructing a `MemoryFileSystem`, you can pass an optional argument (which defaults to posix, that is using `/` for path separators) `FileSystemStyle style = FileSystemStyle.posix` https://pub.dev/documentation/file/latest/memory/MemoryFileSystem/MemoryFileSystem.test.html

**Review by thkim1011 — APPROVED** [review]
LGTM! Thanks so much for your contributions!

File: packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexV525) [reply packages/flutter_tools/test/commands.shard/hermetic/generate_localizations_test.dart:314]: > When constructing a `MemoryFileSystem`, you can pass an optional argument

The problem is when the host platform is Windows, `Uri.toFilePath` will convert the config to `lib\l10n\untranslated.json` which uses a Windows style. If we do use a Windows style here for the file system, the test will be failed too on platforms other than Windows.

**Pull request merged**